br/pkg/streamhelper/advancer.go

```diff
@@ -405,10 +400,15 @@ func (c *CheckpointAdvancer) importantTick(ctx context.Context) error {
 	}
 	p, err := c.env.BlockGCUntil(ctx, c.lastCheckpoint-1)
 	if err != nil {
-		return errors.Annotate(err, "failed to update service GC safe point")
+		return errors.Annotatef(err,
+			"failed to update service GC safe point, current checkpoint is %d, target checkpoint is %d",
+			c.lastCheckpoint-1, p)
+	}
+	if p <= c.lastCheckpoint-1 {
+		log.Info("updated log backup GC safe point.", zap.Uint64("checkpoint", p), zap.Uint64("target", c.lastCheckpoint-1))
 	}
-	if p >= c.lastCheckpoint-1 {
-		log.Info("updated log backup GC safe point.", zap.Uint64("checkpoint", p))
+	if p > c.lastCheckpoint-1 {
+		log.Warn("update log backup GC safe point failed: stale.", zap.Uint64("checkpoint", p), zap.Uint64("target", c.lastCheckpoint-1))
```

Comment:
when will this happened?

Reply:
When the current checkpoint TS is lower than the GC safepoint (which is rare).

Reply:
Will it lost data here when the current checkpoint TS is lower than the GC safepoint?

Reply:
Probably. Perhaps we need to issue some fatal error then. 🤔

Reply:
(Maybe in another PR we should introduce TiDB-side fatal errors.)